src/core/core/task.cpp

```diff
@@ -201,10 +210,12 @@ void task::exec_internal()
         {
```

Comment:
here, can we assert "RUNNING_STATE == TASK_STATE_READY"? because if _state is not RUNNING, it must be READY (changed by timer or reset_retry).